03_letters/hsy3418/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var letterTests = []struct {
+	input string
+	want  map[rune]int
+}{
+	{input: "abcdefghijk",
+		want: map[rune]int{'a': 1, 'b': 1, 'c': 1, 'd': 1, 'e': 1, 'f': 1, 'g': 1, 'h': 1, 'i': 1, 'j': 1, 'k': 1}},
+	{input: "894431kdk",
+		want: map[rune]int{'8': 1, '9': 1, '4': 2, '3': 1, '1': 1, 'k': 2, 'd': 1}},
+}
+
+var sortletterTests = []struct {
+	input map[rune]int
+	want  []string
+}{
+	{input: map[rune]int{'a': 1, 'b': 1, 'c': 1, 'd': 1, 'e': 1, 'f': 1, 'g': 1, 'h': 1, 'i': 1, 'j': 1, 'k': 1},
+		want: []string{"a:1", "b:1", "c:1", "d:1", "e:1", "f:1", "g:1", "h:1", "i:1", "j:1", "k:1"}},
+	{input: map[rune]int{'8': 1, '9': 1, '4': 2, '3': 1, '1': 1, 'k': 2, 'd': 1},
+		want: []string{"1:1", "3:1", "4:2", "8:1", "9:1", "d:1", "k:2"}},
+}
+
+func TestLetters(t *testing.T) {
+	for _, test := range letterTests {
+		actual := letters(test.input)
+		require.Equal(t, test.want, actual)
```

Comment:
Why use require which will ignore the rest test cases when one test case fails? Why not consider using assert from github.com/stretchr/testify/assert